deep_qa/layers/backend/collapse_to_batch.py

```diff
@@ -0,0 +1,65 @@
+from overrides import overrides
+import tensorflow as tf
+
+from ..masked_layer import MaskedLayer
+
+
+class CollapseToBatch(MaskedLayer):
+    """
+    Reshapes a higher order tensor, taking the first ``num_to_collapse`` dimensions and folding
+    them into the batch dimension.  For example, a tensor of shape (2, 4, 5, 3), collapsed with
+    ``num_to_collapse = 2``, would become a tensor of shape (40, 3).  We perform identical
```

Comment:
Oh, you mean excluding the original batch size dimension? May be you should mention that.